src/client/entity-editor/relationship-editor/relationship.tsx

```diff
@@ -74,29 +74,42 @@ function Relationship({
 	if (typeof depth !== 'undefined') {
 		indentationClass = `margin-left-d${8 * depth}`;
 	}
-
+	const child =
+	(
+		<div aria-label={description} className={indentationClass}>
+			<Entity data={sourceObject}/>
+			{` ${usedLinkPhrase} `}
+			<Entity data={targetObject}/>
+			{' '}
+			<RelationshipAttribute attributes={attributes} showAttributes={showAttributes}/>
+		</div>
+	);
+	let parentContainer;
+	if (props.getStyles) {
+		parentContainer = props.innerProps ? <components.Option {...props as OptionProps<any, any>}> {child}</components.Option> : props.getStyles &&
+		<components.SingleValue {...props}>{child}</components.SingleValue>;
+	}
```

Comment:
I can't tell what that does, can you explain?
It would be good to have some comments here.

I'm not sure we should be tying this component to react-select, since it doesn't use it directly
, and it is also use with nothing to do with react-select (for example the EntityRelationships component)

Where are props.getStyles and props.innerProps from? They're not mentioned anywhere else in the component

Reply:
Sorry for the mess, these are the [props](https://react-select.com/props#components) react-select pass to custom components. since we are using same component for value, option and other purposes i have to do this to ensure it work in all cases. i will add some comments about the same

Reply:
To me it looks like we shouldn't be using this component directly, if all these changes are required in order to make it work for use in the select.
The reason is that this component isn't specific to react-select and is used in other contexts. It shouldn't have to know anything about react-select.

Even with the comment you added, I can't tell what is happening in this code block because I don't know how the new version of react-select works; that's a red flag: we're tying this component to a specific version of a library.


I think it is probably time to have a RelationshipOption component specifically for use with react-select, that will manage all these implementation details and rely on the unmodified Relationship component internally for displaying.

Does that seem reasonable to you? As I said I'm unclear on react-select details, but happy to help.